preprocessy/feature_selection/_selectKBest.py

```diff
@@ -81,43 +120,35 @@ def __get_mask(self):
         return mask
 
     def fit(self, params):
-        """Function that fits the scoring function over (X,y) and generates the scores and pvalues for all features with the
+        """Function that fits the scoring function over (X_train, y_train) and generates the scores and pvalues for all features with the
         target label. If no scoring function is passed, then defaults to f_classify or f_regression based on the predictive
         problem.
 
         Parameters
         ----------
 
-        X : pd.core.frame.DataFrame of shape (n_samples, n_features). The training input samples.
+        X : pd.core.frame.DataFrame of shape (n_samples, n_features). The training input features.
 
         y : pd.core.series.Series of shape (n_samples,). The target values (class labels in classification, real numbers in
             regression).
 
-        Returns
-        -------
-        self : object
-
         """
-        if "X" in params.keys():
-            self.X = params["X"]
-        if "y" in params.keys():
-            self.y = params["y"]
+
+        if "train_df" in params.keys():
+            self.train_df = params["train_df"]
+        if "test_df" in params.keys():
+            self.test_df = params["test_df"]
+        if "target_col" in params.keys():
+            self.target_col = params["target_col"]
         if "score_func" in params.keys():
             self.score_func = params["score_func"]
         if "k" in params.keys():
             self.k = params["k"]
 
-        if not isinstance(self.X, pd.core.frame.DataFrame):
-            raise TypeError(
-                "Feature dataframe is not a valid dataframe.\nExpected object"
-                " type: pandas.core.frame.DataFrame"
-            )
+        self.__validate_input()
 
-        if not isinstance(self.y, pd.core.series.Series):
-            raise TypeError(
-                "Target column is not a valid series.\nExpected object type:"
-                " pandas.core.series.Series"
-            )
+        self.X = self.train_df.drop(self.target_col, axis=1)
+        self.y = self.train_df[self.target_col]
 
         if self.score_func is None:
             if self.y.nunique() <= 15:
```

Comment:
The 15 here is to check whether classification or regression right?

Reply:
If it is won't it be better to keep this threshold as some percent of the total rows? Say nunique/total_rows <= 0.2 maybe?

Reply:
We can. Let me know a good threshold and I'll update this.